File: src/ios/FirebasePluginMessageReceiverManager.m
#import "FirebasePluginMessageReceiverManager.h"

@implementation FirebasePluginMessageReceiverManager

static NSMutableArray* receivers;

+ (void) register:(FirebasePluginMessageReceiver*)receiver {
    if(receivers == nil){
        receivers = [[NSMutableArray alloc] init];
    }
    [receivers addObject:receiver];
}

+ (bool) sendNotification:(NSDictionary *)userInfo {
    bool handled = false;
    for(FirebasePluginMessageReceiver* receiver in receivers){
        bool wasHandled = [receiver sendNotification:userInfo];
        if(wasHandled){
            handled = true;
        }
    }
    return handled;
}
@end

File: src/android/FirebasePluginMessageReceiverManager.java
package org.apache.cordova.firebase;

import android.os.Bundle;

import com.google.firebase.messaging.RemoteMessage;

import java.util.ArrayList;
import java.util.List;

public class FirebasePluginMessageReceiverManager {

    private static List<FirebasePluginMessageReceiver> receivers = new ArrayList<FirebasePluginMessageReceiver>();

    public static void register(FirebasePluginMessageReceiver receiver) {
        receivers.add(receiver);
    }

    public static boolean onMessageReceived(RemoteMessage remoteMessage) {
        boolean handled = false;
        for (FirebasePluginMessageReceiver receiver : receivers) {
            boolean wasHandled = receiver.onMessageReceived(remoteMessage);
            if (wasHandled) {
                handled = true;
            }
        }

        return handled;
    }

    public static boolean sendMessage(Bundle bundle) {
            boolean handled = false;
            for (FirebasePluginMessageReceiver receiver : receivers) {
                boolean wasHandled = receiver.sendMessage(bundle);
                if (wasHandled) {
                    handled = true;
                }
            }

            return handled;
        }
}
